src/device-manager/java/src/nl/Weave/DataManagement/GenericTraitUpdatableDataSink.java

```diff
@@ -340,6 +340,7 @@
      * that has been assigned via {@link #setCompletionHandler}.
      */
     public void beginRefreshData();
+    public void beginRefreshDataWithEvents();
```

Comment:
I'm not quite sure of the implementation details here, but overall I'm curious if we even need to tie this to refreshData? Could this just be a separate getEvents() command, entirely separate from the trait refresh? That avoids the need for these two parallel RefreshData / RefreshDataWithEvents methods.

Reply:
Changed, beginRefreshDataWithEvents -> beginFetchEvents(timeout)
And setEventFetchingTimeout is removed